python/mxnet/ndarray/ndarray.py

```diff
@@ -1285,6 +1285,16 @@ def broadcast_to(self, shape):
             return op.broadcast_to(self, shape=tuple(shape))
     # pylint: enable= undefined-variable
 
+    def check_format(self, full_check=True):
```

Comment:
Not sure about adding this to `NDArray` class. I suggest `BaseSparseNDArray` in `sparse.py` instead since for dense there's nothing to check.